fairtally/cli.py

```diff
@@ -1,68 +1,47 @@
-              help="Filename of where to write the results as JSON.",
-              default=None, type=click.File("wt"))
+@click.option("--output-file", "-o", "output_filename",
+              help="Filename of where to write the results. Use `-` to write to standard out.",
+              default="tally.html", show_default=True)
 @click.option("--input-file", "-i", "input_file",
               help="Check URLs in file. One URL per line. Use `-` to read from standard input.",
-              default=None, type=click.File('r'))
-def cli(urls=None, output_file_html=None, output_file_json=None, input_file=None):
+              default=None,
+              type=click.File('rt'))
+@click.option("--format", "output_format",
+              help="Format of output",
+              default="html", show_default=True,
+              type=click.Choice(("html", "json")))
+def cli(urls, input_file, output_format, output_filename):
     all_urls = merge_urls(urls, input_file)
 
     if len(all_urls) == 0:
-        print("No URLs provided, aborting.")
-        return
-
-    results = list()
+        click.echo("No URLs provided, aborting.", err=True)
+        sys.exit(1)
 
     url_progressbar = tqdm(all_urls, bar_format="fairtally progress: |{bar}| {n_fmt}/{total_fmt}", ncols=70, position=0)
     current_value = tqdm(total=0, bar_format="{desc}", position=1)
-    for url in url_progressbar:
-        stderr_buffer = io.StringIO()
-        stdout_buffer = io.StringIO()
-        with RedirectStdStreams(stdout=stdout_buffer, stderr=stderr_buffer):
-            try:
-                current_value.set_description_str("currently checking " + url)
-                repo = Repo(url)
-                compliance = Checker(repo, ignore_repo_config=True, is_quiet=True).check_five_recommendations()
-            except Exception:
-                compliance = Compliance(False, False, False, False, False)
-            finally:
-                badge = "https://img.shields.io/badge/fair--software.eu-{0}-{1}"\
-                        .format(compliance.urlencode(), get_badge_color(compliance))
-                d = dict(url=url, badge=badge, repository=compliance.repository, license=compliance.license,
-                         registry=compliance.registry, citation=compliance.citation, checklist=compliance.checklist,
-                         count=compliance.count(), stdout=stdout_buffer.getvalue(), stderr=stderr_buffer.getvalue())
-
-        current_value.set_description_str()
-        results.append(d)
-
-    if output_file_json is None and output_file_html is None:
-        print(json.dumps(results))
-
-    if output_file_json is not None:
-        write_as_json(results, output_file_json)
-
-    if output_file_html is not None:
-        write_as_html(results, output_file_html)
-
-
-if __name__ == "__main__":
-    cli()
+    results = [check_url(url, current_value) for url in url_progressbar]
+
+    if output_filename == 'tally.html' and output_format == 'json':
```

Comment:
Can we use a CONSTANT instead of a hard-coded filename?

Reply:
done